### PR TITLE
Made a proper img2img alt pipeline 

### PR DESCRIPTION
**Pull request by neonsecret**
Also moved duplicate code to a separate file

### CODE REVIEW

**Comment by hl475** [comment]
@terrychenism can you take a look?

**Comment by hlky** [comment]
I'll have a PR for AITemplate VAE encode soon. I'm working on it [here](https://github.com/hlky/AIT/commit/a5e35971dd796d8ff261fa0a18d34ba4a92f3d94)

**Comment by neonsecret** [comment]
> I'll have a PR for AITemplate VAE encode soon. I'm working on it [here](https://github.com/hlky/AIT/commit/a5e35971dd796d8ff261fa0a18d34ba4a92f3d94)

this one works already, are there any issues?

**Comment by hlky** [comment]
No the pipeline changes and deduplication look good to me, but it's using PyTorch VAE encode because AITemplate VAE encode wasn't implemented. I have implemented VAE encode for AITemplate, it is working in the linked repo. I can wait until this PR is merged then adjust the img2img pipeline to use AITemplate VAE encode

**Comment by neonsecret** [comment]
> No the pipeline changes and deduplication look good to me, but it's using PyTorch VAE encode because AITemplate VAE encode wasn't implemented. I have implemented VAE encode for AITemplate, it is working in the linked repo. I can wait until this PR is merged then adjust the img2img pipeline to use AITemplate VAE encode

yeah that'd be best, thx
but also it's not critical to use vanilla vae encode as it's a one-time thing and not critical on the speed

**Comment by neonsecret** [comment]
> Thanks @neonsecret @hlky for the contribution. Is there any concerns for VAE encode? let me know if you need any help. @hlky
> 
> @neonsecret could you please resolve the conflicts? thanks!

what are the conflicts? I'm not sure what's wrong

**Comment by hlky** [comment]
@terrychenism Only issue is with `DiagonalGaussianDistribution` part of VAE encode, [here](https://github.com/huggingface/diffusers/blob/716286f19ddd9eb417113e064b538706884c8e73/src/diffusers/models/vae.py#L330).
In AITemplate graph it would be like [this](https://github.com/hlky/AIT/blob/a5f1c06629de521f53d2509999041b8b7fbecffa/AITemplate/ait/modeling/vae.py#L214-L262). 
However when using dynamic shape `x = self.mean + self.std * sample` results in `Input shapes of the elementwise op are not compatible!` which comes from [here](https://github.com/facebookincubator/AITemplate/blob/8992f3f2829a0f81a117b1f97aad5e369606a1d7/python/aitemplate/compiler/ops/common/elementwise.py#L80).
I'm not sure if elementwise is not compatible with IntVar, or if there is another solution.
I encountered the same issue with elementwise when trying to get dynamic shape working for ControlNet modules.
For VAE encode it can be worked around by doing the [sample in PyTorch](https://github.com/hlky/AIT/blob/a5f1c06629de521f53d2509999041b8b7fbecffa/AITemplate/AITemplate.py#L346-L347).
If you have any insight it would be appreciated.

**Comment by hlky** [comment]
@terrychenism 

The above issue was fixed [here](https://github.com/hlky/AIT/commit/b7f96653e941c7c22223cc7b3b2d1e97252ee576#diff-a2a33902550ace757cb7207591c0f10fa7e8c8dc82d3a118ab97b4efccfaf9deR284-R285), so the full VAE encode now happens in AITemplate.

Setting the shape of one to the shape of the other also fixed the elementwise issue for [ControlNet dynamic shape](https://github.com/hlky/AIT/commit/87679144fea9f3b1c600b7fcfebf38136cb5d264#diff-01a1a4195c01eae928bb51e3c276068a959522cb4ee6eb53bbbe2b94f50a48d6R215).

When this PR is merged I am ready to bring VAE encode upstream.

File: examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py [hunk 1]
```diff
@@ -13,532 +13,43 @@
...
 from diffusers.utils.pil_utils import numpy_to_pil
 from tqdm import tqdm
-
 from transformers import CLIPTextConfig, CLIPTextModel, CLIPTokenizer
 
 from .compile_lib.compile_vae_alt import map_vae_params
 from .modeling.vae import AutoencoderKL as ait_AutoencoderKL
+from .pipeline_utils import convert_ldm_unet_checkpoint, convert_ldm_vae_checkpoint
 
+from diffusers import (
```
Comment (ipiszy) [inline examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34]: Please fix lint errors.

**Review by ipiszy — COMMENTED** [review]
Thanks @neonsecret!

Please rebase and fix merge conflicts in 
examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py
examples/05_stable_diffusion/src/pipeline_stable_diffusion_controlnet_ait.py

File: examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py (hunk 1 above)
Comment (neonsecret) [reply examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34]: What's wrong? Can't quite figure out

Comment (hlky) [reply examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34]: Check the log of [lint action](https://github.com/facebookincubator/AITemplate/actions/runs/5247622075/jobs/9627067379?pr=762)
```
Error: ./examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34:1: F401 'diffusers.DDIMScheduler' imported but unused
Error: ./examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34:1: F401 'diffusers.PNDMScheduler' imported but unused
Error: ./examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34:1: F401 'diffusers.EulerDiscreteScheduler' imported but unused
Error: ./examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:789:13: F841 local variable 'offset' is assigned to but never used
```

Comment (neonsecret) [reply examples/05_stable_diffusion/src/pipeline_stable_diffusion_ait_alt.py:34]: oh thanks, gonna fix

**Comment by neonsecret** [comment]
OK what's wrong again

**Comment by ipiszy** [comment]
> OK what's wrong again

There are some merge conflicts, please rebase?

Screenshot 2023-06-20 at 4 55 52 PM

**Comment by neonsecret** [comment]
I will redo this cleanly

**Pull request closed without merging**